### PR TITLE
Elvijs/make releases easy

### PR DESCRIPTION
**Pull request by elvijs**
This PR adds a GitHub workflow to enable easy release process.

The process is:
* bump up the version of the package in `setup.py`,
* create a release via GitHub,
* profit! 

I did a test against an older commit. After creating [a release via GitHub UI](https://github.com/ManchesterBioinference/BranchedGP/releases/tag/v0.1a2), the workflow added in this PR automatically created [this build](https://github.com/ManchesterBioinference/BranchedGP/actions/runs/287932692), which in turn published [this release](https://test.pypi.org/project/BranchedGP/0.1a2/) to the test. You can check the availability of the package via `python3 -m pip install --index-url https://test.pypi.org/simple/ --no-deps BranchedGP`. Note that Tensorflow doesn't upload their packages to the test PyPI, so we cannot check actual package functionality via the test index.

Once this is merged in, we will be able to easily release a new version (maybe we could make it version 1.0?). I'll then also update the BGP wiki.

### CODE REVIEW

File: .github/workflows/publish-package.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Publish BranchedGP to TestPyPI and PyPI
+on: push
+
+jobs:
+  publish:
+
+    runs-on: ubuntu-18.04
+
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access ita
```
Comment (alexisboukouvalas) [inline .github/workflows/publish-package.yml:10]: typo - remote a from its

**Review by alexisboukouvalas — APPROVED** [review]
nice! Could you add somewhere (e.g. readme?) how to make releases? For myself, I am not familiar with the second step - where do I go on the GitHub UI to make a release?

**Comment by elvijs** [comment]
> nice! Could you add somewhere (e.g. readme?) how to make releases? For myself, I am not familiar with the second step - where do I go on the GitHub UI to make a release?

Sure thing! How about using the wiki instead? Releases seem like a project management detail, so maybe worth keeping away from the package docs

**Comment by elvijs** [comment]
> nice! Could you add somewhere (e.g. readme?) how to make releases? For myself, I am not familiar with the second step - where do I go on the GitHub UI to make a release?

Added the release instructions to https://github.com/ManchesterBioinference/BranchedGP/wiki

**Pull request merged**